test/e2e/managed_resource_vap_test.go

```diff
@@ -0,0 +1,217 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+
+	testutils "go.goms.io/fleet/test/e2e/v1alpha1/utils"
+)
+
+const (
+	managedByLabel      = "fleet.azure.com/managed-by"
+	managedByLabelValue = "arm"
+	vapName             = "aks-fleet-managed-by-arm"
+	vapBindingName      = "aks-fleet-managed-by-arm"
+)
+
+// Helper functions for creating managed resources
+func createManagedNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				managedByLabel: managedByLabelValue,
+			},
+		},
+	}
+}
+
+func createUnmanagedNamespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+func createManagedResourceQuota(name, namespace string) *corev1.ResourceQuota {
+	return &corev1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				managedByLabel: managedByLabelValue,
+			},
+		},
+		Spec: corev1.ResourceQuotaSpec{
+			Hard: corev1.ResourceList{
+				corev1.ResourcePods: resource.MustParse("10"),
+			},
+		},
+	}
+}
+
+func createManagedNetworkPolicy(name, namespace string) *networkingv1.NetworkPolicy {
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				managedByLabel: managedByLabelValue,
+			},
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+			},
+		},
+	}
+}
+
+func createManagedCRP(name string) *placementv1beta1.ClusterResourcePlacement {
+	return &placementv1beta1.ClusterResourcePlacement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				managedByLabel: managedByLabelValue,
+			},
+		},
+		Spec: placementv1beta1.PlacementSpec{
+			ResourceSelectors: []placementv1beta1.ClusterResourceSelector{
+				{
+					Group:   "",
+					Version: "v1",
+					Kind:    "Namespace",
+					Name:    "test-ns",
+				},
+			},
+		},
+	}
+}
+
+func expectDeniedByVAP(err error) {
```

Comment:
https://onsi.github.io/ginkgo/#testing-external-systems We need pass g Gomega var and use g.Expect inside Eventually blocks https://github.com/Azure/fleet/blob/main/test/e2e/webhook_test.go#L124